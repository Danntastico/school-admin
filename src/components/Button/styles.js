import styled from "styled-components";

export const Btn = styled.button`
  color: var(--secondary-color);
  background-color: ${(props) => props.inputColor || "var(--yellow-color)"};
  border-radius: 20px;
  font-size: 12px;
  font-weight: bold;
  height: 42px;
  text-transform: uppercase;
  min-width: 176px;
  transition: 0.4s;
  opacity: 0.9;
  cursor: pointer;
  :hover {
    transform: translate(0, -5px);
    box-shadow: 0px 5px 5px 0px rgba(0, 0, 0, 0.19);
  }
  &[disabled] {
    opacity: 0.4;
  }
`;
