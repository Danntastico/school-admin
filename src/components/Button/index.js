import React from "react";
import { Btn } from "./styles";
import PropTypes from "prop-types";

export const Button = ({ onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
