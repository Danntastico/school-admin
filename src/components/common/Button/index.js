import React from "react";
import { Btn } from "./styles";

export const Button = ({ type, onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} type={type} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
