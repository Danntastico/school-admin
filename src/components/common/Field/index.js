import React from "react";
import { Input, Label, FieldContainer } from "./styles";
export const Field = ({
  disabled = false,
  fieldType,
  handleInputChange,
  inputType,
  label,
  name,
  placeholder,
  value,
  children,
  min,
  max,
}) => {
  const fieldRender = () => {
    switch (fieldType) {
      case "textarea":
        return (
          <textarea
            autoComplete="off"
            disabled={disabled}
            name={name}
            id={name}
            onChange={handleInputChange}
            value={value}
          />
        );
      case "select":
        return (
          <select
            name={name}
            value={value}
            required={true}
            onChange={handleInputChange}
          >
            {children}
          </select>
        );
      default:
        return (
          <Input
            autoComplete="off"
            disabled={disabled}
            id={name}
            name={name}
            onChange={handleInputChange}
            placeholder={placeholder}
            type={inputType}
            value={value}
            step="0.01"
            required={true}
            min={min}
            max={max}
          />
        );
    }
  };
  return (
    <FieldContainer>
      <Label htmlFor={name}> {label}</Label>
      {fieldRender()}
    </FieldContainer>
  );
};
