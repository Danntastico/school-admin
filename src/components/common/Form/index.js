import React from "react";
import { MdEdit } from "react-icons/md";
import { FormHeader, FormTitle, FormBody, FormFooter, EditBtn } from "./styles";

export const Form = ({
  children,
  hideHeader,
  title,
  onSubmit,
  isEditable,
  onClick,
}) => {
  const editBtn = () => {
    if (isEditable)
      return (
        <EditBtn onClick={onClick}>
          <MdEdit />
        </EditBtn>
      );
  };
  return (
    <form onSubmit={onSubmit}>
      {!hideHeader && (
        <FormHeader>
          <FormTitle> {title} </FormTitle>
          {editBtn()}
        </FormHeader>
      )}
      <FormBody>{children[0]}</FormBody>
      <FormFooter>{children[1]}</FormFooter>
    </form>
  );
};
