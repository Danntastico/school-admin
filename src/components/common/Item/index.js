import React from "react";
import { ItemList } from "./styles";

export const Item = ({ children, bgColor, txtColor, to, onClick }) => (
  <ItemList bgColor={bgColor} txtcolor={txtColor} onClick={onClick}>
    {children}
  </ItemList>
);
