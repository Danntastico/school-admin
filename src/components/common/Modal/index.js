import React, { useRef, useEffect } from "react";
import { Modal as ModalStyled, ModalChildren } from "./styles";
export const Modal = ({ children, modalIsOpen }) => {
  const modalRef = useRef(false);
  useEffect(() => {
    if (modalIsOpen) {
      modalRef.current.style.display = "block";
    } else {
      modalRef.current.style.display = "none";
    }
  }, [modalIsOpen]);
  return (
    <ModalStyled ref={modalRef} className="modal">
      <ModalChildren className="modal__content">{children}</ModalChildren>
    </ModalStyled>
  );
};
