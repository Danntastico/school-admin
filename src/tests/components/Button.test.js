import React from "react";
import "@testing-library/jest-dom";
import { Button } from "../../components/common/Button";
import { mount } from "enzyme";

describe("Unit testing for Button Component", () => {
  const mockOnclick = jest.fn();
  const label = "Test Btn";
  const wrapper = mount(<Button onClick={mockOnclick} label={label} />);

  test("Should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Test Clicks events", () => {
    console.log(wrapper.find("button").simulate("click"));
    expect(mockOnclick.mock.calls.length).toEqual(1);
  });
});
