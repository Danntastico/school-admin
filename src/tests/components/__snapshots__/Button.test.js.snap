// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing for Button Component Should show correctly 1`] = `
<button
  className="sc-AxjAm WiIod"
  onClick={[MockFunction]}
>
  Test Btn
</button>
`;
